### PR TITLE
Move pre-specializations of popular types away from the stdlib

### PR DESCRIPTION
**Pull request by swiftix**
Pre-specializations were only used by Onone builds, but were kept inside the standard library dylyb anyways. This commit moves all the pre-specializations into a dedicated Swift module and a dynamic library, which are only used by Onone builds.

This reduces the code size of libswiftCore.dylib by 5%.

### CODE REVIEW

**Comment by swiftix** [comment]
@swift-ci Please test

File: stdlib/public/OnoneSupport/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+add_swift_library(swiftSwiftOnoneSupport SHARED IS_STDLIB
```
Comment (gribozavr) [inline stdlib/public/OnoneSupport/CMakeLists.txt]: For consistency, this directory should be named 'SwiftOnoneSupport'.

File: stdlib/public/OnoneSupport/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+ # This file should be listed the first.  Module name is inferred from the
+ # filename.
+ SwiftOnoneSupport.swift
+ SWIFT_COMPILE_FLAGS ${STDLIB_SIL_SERIALIZE_ALL}
```
Comment (gribozavr) [inline stdlib/public/OnoneSupport/CMakeLists.txt]: The indentation on these lines should be 2 spaces.

File: lib/SILOptimizer/Utils/Generics.cpp [hunk 3]
```diff
@@ -145,7 +145,7 @@ static bool cacheSpecialization(SILModule &M, SILFunction *F) {
 
   if (M.getOptions().Optimization >= SILOptions::SILOptMode::Optimize &&
       F->getLinkage() != SILLinkage::Public &&
-      F->getModule().getSwiftModule()->getName().str() == STDLIB_NAME) {
+      F->getModule().getSwiftModule()->getName().str() == "SwiftOnoneSupport") {
```
Comment (gribozavr) [inline lib/SILOptimizer/Utils/Generics.cpp]: Probably this magic module name should be factored into a constant like `STDLIB_NAME`.

**Comment by swiftix** [comment]
@gribozavr Thanks for the review, Dmitri! I've incorporated your suggested changes and will push-force the corrected version on my branch.

BTW, do you have any idea why OS X tests pass and Linux tests fail? Do we test PM only on Linux?

**Comment by gribozavr** [comment]
@swiftix It looks like we run swiftpm tests on OS X (https://ci.swift.org/job/swift-PR-osx/173/console#console-section-9).  Maybe that is a smaller set of tests?  I don't know.

**Comment by swiftix** [comment]
@swift-ci Please test

**Comment by swiftix** [comment]
@gribozavr Yes, indeed. We run PM tests on OS X. I wonder if may be it is related to using different linkers on OS X and Linux. Or may be auto-linking works differently?

**Comment by gribozavr** [comment]
Autolinking definitely uses a different mechanism.  On OS X, we are using a linker feature.  On Linux, we emit the information into a special section, and then, when linking with `swift`, we first scan the contents of the sections of inputs for linker flags.

**Comment by swiftix** [comment]
@swift-ci Please test

**Comment by swiftix** [comment]
This is currently blocked by the following issue: https://github.com/apple/swift-corelibs-xctest/commit/6a1e7836e91940be94856bfda245deb125a66729

**Comment by swiftix** [comment]
@swift-ci Please test

**Pull request merged**